spirv_glsl.cpp

```diff
@@ -3968,7 +3994,10 @@ string CompilerGLSL::to_function_name(uint32_t, const SPIRType &imgtype, bool is
 		fname += "texelFetch";
 	else
 	{
-		fname += "texture";
+		if (is_legacy() && ((imgtype.basetype == SPIRType::SampledImage) || (imgtype.basetype == SPIRType::Sampler)) && imgtype.image.depth)
```

Comment:
Why SPIRType::Sampler?

Reply:
@HansKristian-ARM I copy and pasted the if more or less verbatim from line 8480. Same goes for the other instance.  Based on the comment I guess the check is unnecessary? Thinking about this, just checking for depth probably ought to be enough here?

Reply:
I think just checking for depth is enough, either that, or check if the op-code has "Dref" in it.